src/garage/tf/algos/ppo.py

```diff
@@ -77,6 +81,7 @@ def __init__(self,
                  stop_entropy_gradient=False,
                  entropy_method='no_entropy',
                  flatten_input=True,
+                 fit_baseline_order='after',
```

Comment:
what inheritance hierarchy do you plan for RL2/RL2PPO/RL2TRPO?

it is dangerous for this to be a constructor option in PPO when there is no reason for anyone to use it except for in RL2.

Reply:
I plan to have RL2PPO inherit RL2NPO, which inherits NPO by overriding `fit_baseline_before`. The conflicting part here is that if it isn't for RL2, NPO won't need the separation of `fit_baseline_before` and `fit_baseline_after`, so if you look at NPO alone separately, the refactor is weird and unnecessary.

How about we enforce PPO to have `fit_baseline_order='after'` while provides both options for RL2PPO/RL2TRPO?

Reply:
see my comments above. it makes no sense to have methods `fit_baseline_before` and `fit_baseline_after` -- there is only one `fit_baseline` and what you want to control is when it's called.

I think your inheritance tree should be RL2NPO <-- NPO, RL2PPO <-- RL2NPO, RL2TRPO <-- RL2NPO. That way you can specify `fit_baseline='after'` for TRPO and PPO and allow it to be an option for RL2PPO and RL2TRPO.